### ISSUE TITLE
Suggestion for finding more Visual Studio build numbers.

### ISSUE THREAD

**Issue by MikeStratoti** [dishather/richprint#2]
Thank you for maintaining the comp_id.txt list!

It seems that Microsoft maintains official lists of recent Visual Studio releases at:
https://learn.microsoft.com/en-us/visualstudio/install/visual-studio-build-numbers-and-release-dates?view=vs-2022
https://learn.microsoft.com/en-us/visualstudio/install/visual-studio-build-numbers-and-release-dates?view=vs-2019
https://learn.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2017/install/visual-studio-build-numbers-and-release-dates?view=vs-2017

These may help fill in the gaps or verify your interpolations.

**Comment by dishather** [comment]
Oh... Thank you!
That's a lot of build numbers to check and verify, may take me a while.

**Comment by MikeStratoti** [comment]
Do you know about the "/Bv" option to the compiler?
It returns the build number of the build tools (i.e. the 3rd part of the version)
For VS2017, for example,  **cl /Bv** returns
Compiler Passes:
 C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.16.27023\bin\HostX86\x86\cl.exe:        Version 19.16.**27048**.0
 C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\VC\Tools\MSVC\14.16.27023\bin\HostX86\x86\c1.dll:        Version 19.16.27048.0
...

**Comment by dishather** [comment]
Getting the build number from an installed compiler is easy: cl.exe prints its build number along with the copyright on each invocation. That is, unless it was invoked with /nologo. :)

**Comment by dishather** [comment]
Well, that's interesting.
I checked several VS2022 build numbers from the Microsoft page, and they all look... weird.
E.g., current version VS2022 v17.5.1 is said to have build number 33424, but on my computer it's 32215. I checked in VS Community Edition, but I don't think other editions are different re tools build numbers. Or am I wrong?

Other versions I checked:
v17.4.5: 33403 vs 31942 (the list on the site is 1461 builds ahead)
v17.4.3: 33205 vs 31937 (1268 builds ahead)
v17.2.0 pre 3.0: 32408 vs 31326 (1082 builds ahead)
v17.0.0 pre 4.0: 31710 vs 30528 (1182 builds ahead)

What's even more weird, some rather distant versions have the same build number and close release dates:
MSVS2022 v17.0.11 released June 14, 2022 (17.0.32602.201)
MSVS2022 v17.2.4 released June 14, 2022 (17.2.32602.215)

MSVS2022 v17.0.14 released September 13, 2022 (17.0.32901.226)
MSVS2022 v17.2.8 released September 13, 2022 (17.2.32901.213)
MSVS2022 v17.3.4 released September 13, 2022 (17.3.32901.215)

MSVS2022 v17.0.16 released November 8, 2022 (17.0.33027.175)
MSVS2022 v17.2.10 released November 8, 2022 (17.2.33027.314)
MSVS2022 v17.4.0 pre 6.0 released November 1, 2022 (17.4.33027.239)

MSVS2022 v17.0.19 released February 14, 2023 (17.0.33402.176) - really?
MSVS2022 v17.2.13 released February 14, 2023 (17.2.33402.178)
MSVS2022 v17.5.0 pre 6.0 released February 7, 2023 (17.5.33402.96)

I sent my feedback to MS to report the issue (I guess it's an issue after all). Hopefully they will fix it.

**Comment by MikeStratoti** [comment]
Another suggestion to verify tooling versions.  
https://godbolt.org/ provides many versions of MSVC and other compilers on-line.
Using a simple program like (in the left hand window) 
```
#include <iostream>
int main()
    {
    std::cout << _MSC_VER;
    }
```
and add the **/Bv** compiler option (in the upper right hand window).
The output (in the lower right hand window) shows the tooling versions:
```
x86 msvc v19.33 - 2588ms (579982B)

example.cpp
Compiler Passes:
 C:\data\msvc\14.33.31631\bin\Hostx86\x86\cl.exe:        Version 19.33.31631.0
 C:\data\msvc\14.33.31631\bin\Hostx86\x86\c1.dll:        Version 19.33.31631.0
 C:\data\msvc\14.33.31631\bin\Hostx86\x86\c1xx.dll:      Version 19.33.31631.0
 C:\data\msvc\14.33.31631\bin\Hostx86\x86\c2.dll:        Version 19.33.31631.0
 C:\data\msvc\14.33.31631\bin\Hostx86\x86\c1xx.dll:      Version 19.33.31631.0
 C:\data\msvc\14.33.31631\bin\Hostx86\x86\link.exe:      Version 14.33.31631.0
 C:\data\msvc\14.33.31631\bin\Hostx86\x86\mspdb140.dll:  Version 14.33.31631.0
 C:\data\msvc\14.33.31631\bin\Hostx86\x86\1033\clui.dll: Version 19.33.31631.0

Compiler returned: 0
```

Thank you for your efforts.
